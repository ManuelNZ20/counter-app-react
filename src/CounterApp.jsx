import PropTypes from "prop-types";
import { useState } from "react";// es un hook

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = (event) => {
    setCounter(counter + 1);
  };
  const handleRemove = (event) => {
    if (counter > 1 ) setCounter(counter - 1);
  };
  const handleReset = (event) => {
    setCounter(1);
  }
  return (
    <>
      <h1 className="counter">Counter App</h1>
      <h2 className="counter">{counter}</h2>
      <div className="flex-button">

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleRemove}>-1</button>
      <button onClick={handleReset}>Reset</button>
      </div>
    </>
  );
};

CounterApp.propsTypes = {
  value: PropTypes.number.isRequired,
};
