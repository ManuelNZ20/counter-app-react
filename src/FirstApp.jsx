import React from "react";
import PropTypes from 'prop-types'

export const FirstApp = ({ title ,subTitle,name}) => {
  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

// Le agrega un valor por defecto a mis props
FirstApp.defaultProps = {
  title:'No ahí título',
  subTitle:'No ahí numero de sub título',
  name:'Manuel Walter'
}
