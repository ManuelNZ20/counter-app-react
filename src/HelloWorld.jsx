
const HelloWorldApp = () => {
  return (
    <h1 className="mensaje">
        Hello World App
    </h1>
  )
}


export default HelloWorldApp;