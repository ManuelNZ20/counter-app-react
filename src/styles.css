body {
    background-color:rgb(38, 37, 37);
    color:white;
    padding:5px;
    font-family: Helvetica,Arial, sans-serif;
    font-size:1.3rem;
}

.counter {
    text-align: center;
}

button {
    padding:15px;
    background-color: cadetblue;
    font-size: 1.3rem;
    border: none;
    border-radius: 12px;
    box-shadow: 3px 3px black;
}
button:hover {
    background-color: rgba(95, 158, 160, 0.748);
    box-shadow: 0px 0px black;
}
button:active {
    background-color: rgba(69, 113, 115, 0.748);
}

.flex-button {
    display: flex;
    flex-direction: row;
    gap: 10px;
    justify-content: center;
}
