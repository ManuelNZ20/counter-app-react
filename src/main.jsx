import React from "react";
import ReactDOM from "react-dom/client";
import { FirstApp } from "./FirstApp";
import "./styles.css";
import { CounterApp } from "./CounterApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <FirstApp title="Hola Mundo" subTitle={'Soy un subtÃ­tulo'}/> */}
    <CounterApp
            value={1}
        > 
        </CounterApp>
       
  </React.StrictMode>
);
