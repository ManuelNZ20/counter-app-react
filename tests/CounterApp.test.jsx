// # Tarea
// # Pruebas en el <CounterApp />
// # test: debe de hacer match con el snapshot
// # test: debe de mostrar el valor inicial de 100 <CounterApp value={100}>

import { screen, render, fireEvent } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Pruebas en el <CounterApp />', () => { 
    const value = 1
    test('debe de hacer match con el snapshot', () => { 
        const {container} = render(<CounterApp value={value}></CounterApp>)
        expect(container).toMatchSnapshot()
     })

     test('debe de mostrar el valor incial de 100', () => { 
        render(<CounterApp value={value}></CounterApp>)
        expect(screen.getByText(value)).toBeTruthy()
    })
    
    test('debe de incrementar con el boton +1', () => { 
        
        render(<CounterApp value={value}></CounterApp>)
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByText('2') ).toBeTruthy()
        
       })
    test('debe de decrementar con el boton -1', () => { 
        
        render(<CounterApp value={value}></CounterApp>)
        fireEvent.click( screen.getByText('-1') )
        expect( screen.getByText('1') ).toBeTruthy()
        
    })
    
    test('debe de funcionar el botÃ³n de reset', () => { 
        
        render(<CounterApp value={value}></CounterApp>)
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('Reset') )
        // screen.debug()
        fireEvent.click(screen.getByRole('button',{ name:"btn-reset" }))

        expect( screen.getByText('1') ).toBeTruthy()
        
     })
 })