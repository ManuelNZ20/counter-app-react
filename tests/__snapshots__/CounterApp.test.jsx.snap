// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    class="counter"
  >
    Counter App
  </h1>
  <h2
    class="counter"
  >
    1
  </h2>
  <div
    class="flex-button"
  >
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
